### ISSUE TITLE
UniqueSinglingOutQueries works incorrectly

### ISSUE THREAD

**Issue by AwesomeLemon** [statice/anonymeter#35]
UniqueSinglingOutQueries compares queries by sorting the symbols in them.

This leads to false positives: for example, queries 'BMI == 19.56' and 'BMI == 16.95' are considered the same (because they consist of the same symbols), leading to only one of them being used (decided randomly).

This is a real-world example from the Framingham heart study dataset.

**Comment by AwesomeLemon** [comment]
Hi, a fix I implemented and would appreciate your opinion on is the following: when constructing the query, sort the columns by name. Afterwards, check query uniqueness on the entire strings, without sorting the symbols. This way, the issue of having the same query with different order of terms is avoided, as well as my issue of discarding non-duplicate queries consisting of the same symbols.

To give an example: I assume that the symbol sorting is to be able to detect duplicates of the form "(a == 1) && (b == 2)" and "(b == 2) && (a == 1)". If the columns are ordered by name, the second query is never created, and so the duplicate checking can simply compare the strings.

> I am not sure that this will impact the privacy evaluation

It makes some impact on smaller datasets where only a few univariate queries can be created - by discarding these false positive duplicates, the number of queries is further decreased.

**Comment by MatteoGiomi** [comment]
You are right in your assumption, the sorting was to avoid precisely those duplicate queries. Your idea about enforcing a column order could work, one just need to be careful when selecting random attributes but it's definitely doable. If you have the fix and want to prepare contribution we'd be glad to accept it :-)

> It makes some impact on smaller datasets where only a few univariate queries can be created - by discarding these false positive duplicates, the number of queries is further decreased.

Good point!

**Comment by AwesomeLemon** [comment]
Great, then I'll send a pull request shortly!
